diesel/src/migrations/mod.rs

```diff
@@ -108,6 +108,21 @@ pub fn run_pending_migrations_in_directory<Conn>(conn: &Conn, migrations_dir: &P
     run_migrations(conn, all_migrations, output)
 }
 
+/// Compares migrations found in `migrations_dir` to those that have been applied.
+/// Returns a list of pathbufs and whether they have been applied.
+pub fn mark_migrations_in_directory<Conn>(conn: &Conn, migrations_dir: &Path, _: &mut Write)
+    -> Result<Vec<(Option<PathBuf>, bool)>, RunMigrationsError> where
+        Conn: MigrationConnection,
+{
+    let migrations = migrations_in_directory(migrations_dir)?;
+    setup_database(conn)?;
+    let already_run = conn.previously_run_migration_versions()?;
+    Ok(migrations.into_iter().map(|m| {
+        let applied = already_run.contains(&m.version().to_string());
+        (m.file_path().map(|p| p.to_path_buf()), applied)
+    }).collect::<Vec<_>>())
```

Comment:
Can you reorder this a bit?

```rust
let already_run = …;
let migrations = ….into_iter().map(…).collect(); // Can probably infer type from function return type
Ok(migrations)
```

Or, if you want to fail fast when `migrations_in_directory` fails, just leave that on top and shadow the binding with `let migrations = migrations.into_iter…`.